lib/elixir/test/elixir/kernel/warning_test.exs

```diff
@@ -28,6 +28,26 @@ defmodule Kernel.WarningTest do
     purge Sample
   end
 
+  test "unused variable in redefined function in different file" do
+    output = capture_err(fn ->
+      Code.eval_string """
+      defmodule Sample do
```

Comment:
Let's use the supervisor example as a test so we don't need to define a separate fixture file. :)